crates/rolldown/src/hmr/hmr_manager.rs

```diff
@@ -219,10 +219,10 @@ impl HmrManager {
       self.session_span.clone(),
     )?;
 
-    let module_loader_output =
-      module_loader.fetch_modules(vec![], module_infos_to_be_updated).await?;
-
-    self.cache = module_loader_output.cache;
+    let (module_loader_output, cache) =
+      module_loader.fetch_modules(vec![], module_infos_to_be_updated).await;
+    self.cache = cache;
+    let module_loader_output = module_loader_output?;
```

Comment:
`self.cache` was removed in line 217 and was not set here if `module_loader.fetch_modules` errors. That was causing the error. It seems if I set `self.cache` when an error happened, the test works fine. Is this a valid fix?
(the return type of `fetch_modules` doesn't look nice, I'll refactor that if the approach looks good)

Reply:
@IWANABETHATGUY cc